src/Nethermind/Nethermind.JsonRpc/Modules/Eth/EthRpcModule.TransactionExecutor.cs

```diff
@@ -58,7 +65,15 @@ protected TxExecutor(IBlockchainBridge blockchainBridge, IBlockFinder blockFinde
                     return ResultWrapper<TResult>.Fail("Contract creation without any data provided.",
                                                ErrorCodes.InvalidInput);
                 }
-                return ExecuteTx(header.Clone(), tx, cancellationTokenSource.Token);
+
+                return ExecuteTx(clonedHeader, tx, cancellationTokenSource.Token);
+            }
+
+            private static bool ShouldNotSetBaseFee(TransactionForRpc t)
+            {
+                return
+                    (t.GasPrice is null || t.GasPrice == 0) && (t.MaxFeePerGas is null || t.MaxFeePerGas == 0)
+                    && (t.MaxPriorityFeePerGas is null || t.MaxPriorityFeePerGas == 0);
```

Comment:
minor: I would do either: `!(x > 0)`or `x is null or 0`
or maybe better do `ShouldSetBaseFee` and make it `x > 0`

Reply:
changed